astropy/convolution/src/convolve.c

```diff
@@ -198,18 +237,17 @@ FORCE_INLINE void convolve1d_boundary_none(DTYPE * const result,
 #endif
     for (i = wkx; i < nx_minus_wkx; ++i)
     {
-        wkx_minus_i = wkx - i; // wkx - 1
+        wkx_plus_i = wkx + i; // wkx + i
         i_minus_wkx = i - wkx; //i - wkx
-        i_plus_wkx_plus_1 = i + wkx_plus_1; // i + wkx + 1
-        nkx_minus_1_minus_wkx_plus_i = nkx_minus_1 - wkx_minus_i; // nkx - 1 - (wkx - i)
+        i_plus_wkx_plus_1 = wkx_plus_i + 1; // i + wkx + 1
 
         top = 0.;
         if (nan_interpolate) // compile time constant
             bot = 0.;
         {omp_iter_var ii;
         for (ii = i_minus_wkx; ii < i_plus_wkx_plus_1; ++ii)
```

Comment:
I think if we're going to rewrite this, then we should make it even more intuitive. I only left it like this to make it a more direct comparison with the code before. I think we should make it obvious that ``ii`` & ``jj`` are the kernel iterator vars by having ``for (ii = 0, ii < nkx, ++ii)``. This way the kernel is always being accessed transparently. Then dump the mapping into the image array index rather than having both indexes being a result of a visual computation. This gets rid of the need to for ``ker_i`` & ``ker_j`` and I think is easier to follow. Thoughts?

Reply:
@jamienoss - that does sound clearer to me. I'll see what I can do.

Reply:
Pitfall: I was assuming that with ``for (ii = 0, ii < nkx, ++ii)`` the kernel would be accessed as ``g[ii]``. This is however incorrect, see #6264. That would be ``(F*G)(t) = Sum[f(x) * g(x)]`` where ``(F*G)(t) = Sum[f(x) * g(t-x)]`` is needed. So my above suggestion may not be possible or really worth it. We could reverse the kernel before passing through, but that seems like a backstep in performance.

Reply:
Pitfall: I was assuming that with for (ii = 0, ii < nkx, ++ii) the kernel would be accessed as g[ii]. This is however incorrect, see #6264. That would be (F*G)(t) = Sum[f(x) * g(x)] where (F*G)(t) = Sum[f(x) * g(t-x)] is needed. ~~So my above suggestion may not be possible or really worth it. We could reverse the kernel before passing through, but that seems like a backstep in performance.~~

Scrap that, it may still be neater.